lib/Feature/FeatureModelParser.cpp

```diff
@@ -225,4 +252,361 @@ std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)> FeatureModelXmlParser::parseDoc() {
+                                                   xmlFree);
+    FMB.setVmName(std::string(reinterpret_cast<char *>(Cnt.get())));
+  }
+  // After the feature model tag, some metadata is provided.
+  // Currently, we skip the metadata and continue with parsing the feature tree
+
+  for (xmlNode *H = Node->children; H; H = H->next) {
+    if (H->type == XML_ELEMENT_NODE) {
+      if (!xmlStrcmp(H->name, SxfmConstants::FEATURE_TREE)) {
+        // Check whether a custom indentation is defined in the feature tree and
+        // set it accordingly
+        {
+          std::unique_ptr<xmlChar, void (*)(void *)> Cnt(xmlGetProp(H, SxfmConstants::INDENTATION),
+                                                         xmlFree);
+          if (Cnt) {
+            Indentation = reinterpret_cast<char*>(Cnt.get());
+          }
+
+        }
+
+        // Parse the feature tree with all its features and relations among them.
+        if (!parseFeatureTree(xmlNodeGetContent(H))) {
+          return false;
+        }
+
+      // Finally, parse the cross-tree-constraints in the constraints tag.
+      } else if (!xmlStrcmp(H->name, SxfmConstants::CONSTRAINTS)) {
+        if (!parseConstraints(xmlNodeGetContent(H))) {
+          return false;
+        }
+      }
+    }
+  }
+  return true;
+}
+
+std::unique_ptr<FeatureModel> FeatureModelSxfmParser::buildFeatureModel() {
+  std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)> Doc = parseDoc();
+  if (!Doc) {
+    return nullptr;
+  }
+
+  FMB.init();
+  return parseVm(xmlDocGetRootElement(Doc.get())) ? FMB.buildFeatureModel()
+                                                  : nullptr;
+}
+
+std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)> FeatureModelSxfmParser::parseDoc() {
+  // Initialize the XML parser
+  std::unique_ptr<xmlParserCtxt, void (*)(xmlParserCtxtPtr)> Ctxt(
+      xmlNewParserCtxt(), xmlFreeParserCtxt);
+  // Parse the given model by libxml2
+  std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)> Doc(
+      xmlCtxtReadMemory(Ctxt.get(), Sxfm.c_str(), Sxfm.length(), nullptr, nullptr,
+                        XML_PARSE_NOBLANKS),
+      xmlFreeDoc);
+  xmlCleanupParser();
+
+  // In the following, the document is validated.
+  // Therefore, (1) check whether it could be parsed
+  if (Doc && Ctxt->valid) {
+    // (2) validate the sxfm format by using the dtd (document type definition) file
+    xmlValidateDtd(&Ctxt->vctxt, Doc.get(), createDtd().get());
+    if (Ctxt->vctxt.valid) {
+      // TODO (CK):
+      // and (3) check the tree-like structure of the embedded feature model
+      // as well as constraints
+      return Doc;
+    } else {
+      std::cerr << "Failed to validate DTD." << std::endl;
+    }
+  } else {
+    std::cerr << "Failed to parse / validate XML." << std::endl;
+  }
+  return std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)>(nullptr, nullptr);
+}
+
+/// This method checks if the given feature model is valid
+/// \return true iff the feature model is valid
```

Comment:
```suggestion
/// This method checks if the given feature model is valid
///
/// \returns true iff the feature model is valid
```

And please move the docs to the header file.